python/lib/core/dmod/core/dataset.py

```diff
@@ -59,85 +59,93 @@ class Dataset(Serializable):
     Rrepresentation of the descriptive metadata for a grouped collection of data.
     """
 
-    _SERIAL_DATETIME_STR_FORMAT = '%Y-%m-%d %H:%M:%S'
-
-    _KEY_ACCESS_LOCATION = 'access_location'
-    _KEY_CREATED_ON = 'create_on'
-    _KEY_DATA_CATEGORY = 'data_category'
-    _KEY_DATA_DOMAIN = 'data_domain'
-    _KEY_DERIVED_FROM = 'derived_from'
-    _KEY_DERIVATIONS = 'derivations'
-    _KEY_DESCRIPTION = 'description'
-    _KEY_EXPIRES = 'expires'
-    _KEY_IS_READ_ONLY = 'is_read_only'
-    _KEY_LAST_UPDATE = 'last_updated'
-    _KEY_MANAGER_UUID = 'manager_uuid'
-    _KEY_NAME = 'name'
-    _KEY_TYPE = 'type'
-    _KEY_UUID = 'uuid'
-
-    # TODO: move this (and something more to better automatically handle Serializable subtypes) to Serializable directly
-    @classmethod
-    def _date_parse_helper(cls, json_obj: dict, key: str) -> Optional[datetime]:
-        if key in json_obj:
-            return datetime.strptime(json_obj[key], cls.get_datetime_str_format())
-        else:
-            return None
-
-    @classmethod
-    def factory_init_from_deserialized_json(cls, json_obj: dict):
-        try:
-            manager_uuid = UUID(json_obj[cls._KEY_MANAGER_UUID]) if cls._KEY_MANAGER_UUID in json_obj else None
-            return cls(name=json_obj[cls._KEY_NAME],
-                       category=DataCategory.get_for_name(json_obj[cls._KEY_DATA_CATEGORY]),
-                       data_domain=DataDomain.factory_init_from_deserialized_json(json_obj[cls._KEY_DATA_DOMAIN]),
-                       dataset_type=DatasetType.get_for_name(json_obj[cls._KEY_TYPE]),
-                       access_location=json_obj[cls._KEY_ACCESS_LOCATION],
-                       description=json_obj.get(cls._KEY_DESCRIPTION, None),
-                       uuid=UUID(json_obj[cls._KEY_UUID]),
-                       manager_uuid=manager_uuid,
-                       is_read_only=json_obj[cls._KEY_IS_READ_ONLY],
-                       expires=cls._date_parse_helper(json_obj, cls._KEY_EXPIRES),
-                       derived_from=json_obj[cls._KEY_DERIVED_FROM] if cls._KEY_DERIVED_FROM in json_obj else None,
-                       derivations=json_obj[cls._KEY_DERIVATIONS] if cls._KEY_DERIVATIONS in json_obj else [],
-                       created_on=cls._date_parse_helper(json_obj, cls._KEY_CREATED_ON),
-                       last_updated=cls._date_parse_helper(json_obj, cls._KEY_LAST_UPDATE))
-        except Exception as e:
+    _SERIAL_DATETIME_STR_FORMAT: ClassVar = '%Y-%m-%d %H:%M:%S'
+    name: str = Field(description="The name for this dataset, which also should be a unique identifier.")
+    # QUESTION: should this be optional? see factory_init_from_deserialized_json
```

Comment:
@robertbartel, can you please weigh in on this question. See [this](https://github.com/NOAA-OWP/DMOD/pull/239/files#diff-bf7ff51a1703a364c9468cee4a087e6334748b16b1b6347a4f16332079b30786R123-R143) section of code for reference.

Reply:
Not sure I understand. Are you asking if the name should be an optional property?

Reply:
Sorry, I should have been more specific. Should `category` and `data_domain` be Optional? In the previous `__init__`, they were required, non-optional, parameters, but in the previous implementation of `factory_init_from_deserialized_json` both _could_ be `None`.

Reply:
Ah, I see. I believe that was a bug. It would not have made sense to have a dataset without a category or without a domain. I seem to have just left out the sanity check from the deserializer function.

Reply:
Great, thanks! I figured that might make for an interesting dataset.